feat: footer

Chanakya888 opened the pull request.

anton-karlovskiy commented on package.json:
```diff
@@ -32,7 +32,6 @@
     "i18next": "^19.8.4",
     "i18next-browser-languagedetector": "^6.0.1",
     "i18next-http-backend": "^1.0.21",
-    "node-sass": "^4.14.1",
```
Do we drop it?

Chanakya888 replied:
I get a sass error locally if I do

nud3l replied:
@Chanakya888 I think you need to just add this back into the file or revert the change that you made on that file and then hit `yarn install` again to make sure it's also in the lockfile

Chanakya888 replied:
@nud3l I've added it back and verified that it's in yarn.lock file, I get the error which doesn't let me run the app locally 

```
**Failed to compile.**

./src/_general.scss (./node_modules/css-loader/dist/cjs.js??ref--5-oneOf-6-1!./node_modules/postcss-loader/src??postcss!./node_modules/resolve-url-loader??ref--5-oneOf-6-3!./node_modules/sass-loader/dist/cjs.js??ref--5-oneOf-6-4!./src/_general.scss)
Node Sass does not yet support your current environment: Linux 64-bit with Unsupported runtime (88)
For more information on which environments are supported please see:
https://github.com/sass/node-sass/releases/tag/v4.14.1```

anton-karlovskiy commented on src/config/links.ts:
```diff
@@ -19,6 +21,14 @@ const POLKA_BTC_DOC_TREASURE_HUNT_VAULT =
 const POLKA_BTC_DOC_TREASURE_HUNT_RELAYER =
   'https://docs.polkabtc.io/#/start/beta?id=staked-relayer-award-4-dot-for-the-first-50-200-dot-in-total';
 
+const GETTING_STARTED = 'https://docs.polkabtc.io/#/start/overview';
+const VAULTS_DOCS = 'https://docs.polkabtc.io/#/vault/overview';
+const RELAYERS_DOCS = 'https://docs.polkabtc.io/#/relayer/overview';
+const DEVELOPERS_DOCS = 'https://docs.polkabtc.io/#/developers/integration';
+const ROADMAP = 'https://docs.polkabtc.io/#/about/roadmap';
```
Should be capitalized snake_case -> ROAD_MAP.

anton-karlovskiy replied:
Can we follow the naming convention?

anton-karlovskiy commented on src/config/links.ts:
```diff
@@ -19,6 +21,14 @@ const POLKA_BTC_DOC_TREASURE_HUNT_VAULT =
+const ROADMAP = 'https://docs.polkabtc.io/#/about/roadmap';
+
+const PRIVACY_POLICY = 'https://www.interlay.io/docs/privacy-policy.pdf';
+const NEWSLETTER_ACTION = 'https://interlay.us4.list-manage.com/subscribe/post?u=4c3c3f21d3ec4c3ed94ea7353&amp;id=ad217abce9';
```
`NEWSLETTER_ACTION` -> `NEWS_LETTER_ACTION`

anton-karlovskiy replied:
Can be a line break before `export`.

anton-karlovskiy commented on src/config/links.ts:
```diff
@@ -19,6 +21,16 @@ const POLKA_BTC_DOC_TREASURE_HUNT_VAULT =
+const POLKA_BTC_DOC_VAULTS = 'https://docs.polkabtc.io/#/vault/overview';
+const POLKA_BTC_DOC_RELAYERS = 'https://docs.polkabtc.io/#/relayer/overview';
+const POLKA_BTC_DOC_DEVELOPERS = 'https://docs.polkabtc.io/#/developers/integration';
+const POLKA_BTC_DOC_ROAD_MAP = 'https://docs.polkabtc.io/#/about/roadmap';
```
@Chanakya888 
I'm sorry. ROADMAP would make sense here although it should be `road map` according to the dictionary.

anton-karlovskiy commented on src/config/links.ts:
```diff
@@ -19,6 +21,16 @@ const POLKA_BTC_DOC_TREASURE_HUNT_VAULT =
+
+const PRIVACY_POLICY = 'https://www.interlay.io/docs/privacy-policy.pdf';
+const NEWS_LETTER_ACTION =
+'https://interlay.us4.list-manage.com/subscribe/post?u=4c3c3f21d3ec4c3ed94ea7353&amp;id=ad217abce9';
```
leading 2 spaces

anton-karlovskiy commented on src/parts/Footer/footer.module.css:
```diff
@@ -0,0 +1,119 @@
+.footer {
```
@Chanakya888 
I think we should use tailwindcss.

anton-karlovskiy commented on src/parts/Footer/footer.module.css:
```diff
@@ -0,0 +1,119 @@
+    font-weight: 700;
+  }
+  
+  @media only screen and (max-width: 600px) {
```
Breakpoint 600 is not defined in the theme as well.

anton-karlovskiy commented on src/parts/Footer/index.tsx:
```diff
@@ -1,118 +1,315 @@
+const Footer = () => {
+  const { t } = useTranslation();
+
+  return (
```
@Chanakya888 
It looks like `ul` and `li` are not used correctly.
`li` should be direct children.

anton-karlovskiy commented:
@Chanakya888 
Is it OK for me to take this PR and update it?

anton-karlovskiy commented on src/parts/Footer/index.tsx:
```diff
@@ -1,118 +1,315 @@
+  NEWS_LETTER_ACTION
 } from 'config/links';
+import styles from './footer.module.css';
 // TODO: should use an SVG
 import interlayImage from 'assets/img/interlay.png';
 // TODO: should use next-gen format
 import web3FoundationImage from 'assets/img/polkabtc/web3-foundation-grants-badge-black.png';
```
@Chanakya888 
We should use SVGs for logos.
![Capture](https://user-images.githubusercontent.com/49653735/112828160-05b35680-9098-11eb-9c31-79db3e2dc012.PNG)

Chanakya888 commented:
> @Chanakya888
> Is it OK for me to take this PR and update it?

Yes please, this PR has been waiting for a long time

anton-karlovskiy commented:
@Chanakya888 cc @nud3l 
OK, I'll take this PR.
But for now, my focus is on table migration.
After that, I'll handle this PR.

Chanakya888 commented:
> @Chanakya888 cc @nud3l 
> 
> OK, I'll take this PR.
> 
> But for now, my focus is on table migration.
> 
> After that, I'll handle this PR.

Oh, no problem then. I'll quickly address your comments tonight. Do you want me to update to tailwind?

anton-karlovskiy commented:
@Chanakya888 
I wanted to update it as an example.
From now on, we should (must) use `tailwindcss`. It's very helpful and easy to style.

Chanakya888 commented:
> @Chanakya888 
> 
> I wanted to update it as an example.
> 
> From now on, we should (must) use `tailwindcss`. It's very helpful and easy to style.
> 
> 

Alrighty, I'll address the rest of the comments and leave the tailwind for you

Chanakya888 commented:
@anton-karlovskiy I've addressed your comments. I couldn't get the svg of web3 foundation right, I've tried tracing it in illustrator and some online tools, neither of them were successful, I'll ask @alexeiZamyatin for the svg.

nud3l commented:
> @anton-karlovskiy I've addressed your comments. I couldn't get the svg of web3 foundation right, I've tried tracing it in illustrator and some online tools, neither of them were successful, I'll ask @alexeiZamyatin for the svg.

See #ux channel in discord.

anton-karlovskiy commented:
@nud3l cc @Chanakya888 
Although the footer itself is in a good place, there are relatively a lot of TODOs to be done in this PR:
- the footer's layout at the app level
- CSS normalization switch
- quick-fixing broken buttons on the dashboard page

nud3l commented:
New footer looks great. Only thing I'd like to change is that the headings (Contact, Challanges, Feedback, Docs) are bold.

![image](https://user-images.githubusercontent.com/14966470/113887462-77a84000-97fc-11eb-80d0-37204bec7a7f.png)

Not sure where that is in Figma right now, otherwise I would have commented there.

nud3l commented:
Overall, very much like the new footer. One thing to fix before merging this is to not have the footer "floating around" on the landing page:

![image](https://user-images.githubusercontent.com/14966470/114142052-61f66000-994d-11eb-9a25-2b586fee6b3b.png)

nud3l commented on public/index.html:
```diff
@@ -35,7 +35,7 @@
 </head>
 
-<body class='text-primary text-base'>
+<body class='text-textPrimary text-base'>
```
not sure if that is tailwind but the `text-textPrimary` seems quite verbose.

anton-karlovskiy replied:
@nud3l 
Yes, I felt so.
I would like it to be named `text-primary`. But according to the tailwindcss behavior, it will pick up the primary color, not the primary text color.

In my opinion, later if I come up with a better name, we can replace it at that time.
What do you think?

nud3l commented on src/pages/dashboard/dashboard.page.tsx:
```diff
@@ -35,6 +35,7 @@ function Dashboard() {
+        {/* TODO: could remove linkButton */}
         <PolkaBTC linkButton />
         <CollateralLocked linkButton />
         <Collateralization linkButton />
```
I'm not sure where it changed, but the capacity item should be blue and the circle is overflowing from the card:

![image](https://user-images.githubusercontent.com/14966470/114143094-af270180-994e-11eb-9518-d6aca6850a39.png)

anton-karlovskiy replied:
@nud3l 
All of the bugs were fixed.
The color change had been there and it was caused by CSS override due to specificity.
If we only used tailwindcss, this wouldn't happen again.

anton-karlovskiy commented:
@nud3l 
> One thing to fix before merging this is to not have the footer "floating around" on the landing page

That's related to the remaining part (`the footer's layout at the app level`) I mentioned.
The footer layout did not follow the best practice so it's error-prone.
This time, I'm going to fix it but it takes some time. So I'd suggest fixing it on top of the merged codebase.

What do you think? I'll follow you.

nud3l commented:
> @nud3l
> 
> > One thing to fix before merging this is to not have the footer "floating around" on the landing page
> 
> That's related to the remaining part (`the footer's layout at the app level`) I mentioned.
> The footer layout did not follow the best practice so it's error-prone.
> This time, I'm going to fix it but it takes some time. So I'd suggest fixing it on top of the merged codebase.
> 
> What do you think? I'll follow you.

Let's fix it here since we don't have anything incoming in parallel now.

anton-karlovskiy commented on README.md:
````diff
@@ -189,6 +189,14 @@ Test the project.
 yarn test
 ```
 
+### Design System
+
+Visualize the Tailwindcss configuration.
+
+```bash
+yarn tailwind-config-viewer
+```
+
````
@nud3l 
> very nice! Small request: can you add a sentence or two about that in the README with the yarn tailwind-config-viewer command?
maybe after test and on the same hierarchy level under design system or so: https://github.com/interlay/polkabtc-ui#test

Could you please have a look?

anton-karlovskiy commented on src/tailwind.config.js:
```diff
@@ -36,12 +64,35 @@ module.exports = {
+        primary: {
+          lightest: INTERLAY_ROSE[200],
+          lighter: INTERLAY_ROSE[300],
+          light: INTERLAY_ROSE[400],
+          DEFAULT: INTERLAY_ROSE[500],
+          dark: INTERLAY_ROSE[600],
+          darker: INTERLAY_ROSE[700],
+          darkest: INTERLAY_ROSE[800],
+          contrastText: colors.white
+        },
+        secondary: {
+          lightest: INTERLAY_TREE_POPPY[200],
+          lighter: INTERLAY_TREE_POPPY[300],
+          light: INTERLAY_TREE_POPPY[400],
+          DEFAULT: INTERLAY_TREE_POPPY[500],
+          dark: INTERLAY_TREE_POPPY[600],
+          darker: INTERLAY_TREE_POPPY[700],
+          darkest: INTERLAY_TREE_POPPY[800],
+          contrastText: '#333333'
         }
       },
       textColor: theme => ({
-        // TODO: should avoid using complete black (#000) (Re: #212529 in bootstrap)
-        primary: theme('colors.black'),
-        secondary: theme('colors.gray.400')
+        textPrimary: theme('colors.gray.900'),
+        textSecondary: theme('colors.gray.400')
+      }),
+      backgroundColor: theme => ({
+        paper: theme('colors.white'),
+        default: theme('colors.gray.50')
```
@Chanakya888 cc @nud3l 
We have primary & secondary colors and text primary & text secondary colors defined in the theme.
This practice was borrowed from https://material-ui.com/customization/default-theme/ which I think in most cases makes sense.

The primary color is polkadot's pink and the secondary color is bitcoin color for now.

nud3l commented on src/pages/app/app.page.tsx:
```diff
@@ -38,7 +38,7 @@ function Application() {
 
   return (
-    <MainContainer className='text-center white-background min-vh-100 app-page'>
+    <MainContainer className='text-center white-background app-page'>
```
very minor it: can you please make the landing page full view height so that the footer on the landing page is only visible when you scroll down?

nud3l reviewed (commented):
Pending the one comment, LGTM and then ready to merge :)
